### ISSUE TITLE
npm5 not installing a subdependency

### ISSUE THREAD

**Issue by revin** [npm/npm#16744]
#### I'm opening this issue because:

  - [x] npm is producing an incorrect install.

#### What's going wrong?

npm5 is not installing a subdependency `emoji-regex` when I install a direct dependency `github-slugger`.

#### How can the CLI team reproduce the problem?

I'm just doing, from an empty directory:

```sh
$ npm i -ddd github-slugger
```

### npm5 behavior

<details>
<summary>Click for logging output</summary>
<pre>
npm info it worked if it ends with ok
npm verb cli [ '/Users/revinguillen/.nvm/versions/node/v6.10.0/bin/node',
npm verb cli   '/Users/revinguillen/.nvm/versions/node/v6.10.0/bin/npm',
npm verb cli   'i',
npm verb cli   '-ddd',
npm verb cli   'github-slugger@^1' ]
npm info using npm@5.0.0
npm info using node@v6.10.0
npm sill install loadCurrentTree
npm sill install readLocalPackageData
npm http fetch GET 200 https://registry.npmjs.org/github-slugger 359ms
npm sill pacote range manifest for github-slugger@^1 fetched in 397ms
npm sill install loadIdealTree
npm sill install cloneCurrentTreeToIdealTree
npm sill install loadShrinkwrap
npm sill install loadAllDepsIntoIdealTree
npm sill resolveWithNewModule github-slugger@1.1.2 checking installable status
npm http fetch GET 200 https://registry.npmjs.org/emoji-regex 186ms
npm sill pacote range manifest for emoji-regex@^>=6.0.0 <=6.1.1 fetched in 194ms
npm sill resolveWithNewModule emoji-regex@6.1.1 checking installable status
npm sill currentTree tmp
npm sill idealTree tmp
npm sill idealTree └── github-slugger@1.1.2
npm sill install generateActionsToTake
npm sill diffTrees action count 1
npm sill diffTrees add github-slugger@1.1.2
npm sill decomposeActions action count 8
npm sill decomposeActions fetch github-slugger@1.1.2
npm sill decomposeActions extract github-slugger@1.1.2
npm sill decomposeActions preinstall github-slugger@1.1.2
npm sill decomposeActions build github-slugger@1.1.2
npm sill decomposeActions install github-slugger@1.1.2
npm sill decomposeActions postinstall github-slugger@1.1.2
npm sill decomposeActions finalize github-slugger@1.1.2
npm sill decomposeActions refresh-package-json github-slugger@1.1.2
npm sill install executeActions
npm sill doSerial global-install 8
npm verb correctMkdir /Users/revinguillen/.npm/_locks correctMkdir not in flight; initializing
npm verb lock using /Users/revinguillen/.npm/_locks/staging-525484a240850021.lock for /Users/revinguillen/tmp/tmp/node_modules/.staging
npm sill doParallel extract 8
npm sill extract github-slugger@1.1.2
npm sill pacote trying github-slugger@https://registry.npmjs.org/github-slugger/-/github-slugger-1.1.2.tgz by hash: sha1-hmOuyUX0Ta/wckBHxZAVseqF+EY=
npm sill pacote data for sha1-hmOuyUX0Ta/wckBHxZAVseqF+EY= not present. Using manifest.
npm http fetch GET 200 https://registry.npmjs.org/github-slugger/-/github-slugger-1.1.2.tgz 82ms
npm sill pacote github-slugger@https://registry.npmjs.org/github-slugger/-/github-slugger-1.1.2.tgz extracted in 127ms
npm sill doReverseSerial remove 8
npm sill doSerial move 8
npm sill doSerial finalize 8
npm sill finalize /Users/revinguillen/tmp/tmp/node_modules/github-slugger
npm sill doParallel refresh-package-json 8
npm sill refresh-package-json /Users/revinguillen/tmp/tmp/node_modules/github-slugger
npm sill doParallel preinstall 8
npm sill preinstall github-slugger@1.1.2
npm info lifecycle github-slugger@1.1.2~preinstall: github-slugger@1.1.2
npm sill lifecycle github-slugger@1.1.2~preinstall: no script for preinstall, continuing
npm sill doSerial build 8
npm sill build github-slugger@1.1.2
npm info linkStuff github-slugger@1.1.2
npm sill linkStuff github-slugger@1.1.2 has /Users/revinguillen/tmp/tmp/node_modules as its parent node_modules
npm verb linkBins github-slugger@1.1.2
npm verb linkMans github-slugger@1.1.2
npm sill doSerial global-link 8
npm sill doParallel update-linked 8
npm sill doSerial install 8
npm sill install github-slugger@1.1.2
npm info lifecycle github-slugger@1.1.2~install: github-slugger@1.1.2
npm sill lifecycle github-slugger@1.1.2~install: no script for install, continuing
npm sill doSerial postinstall 8
npm sill postinstall github-slugger@1.1.2
npm info lifecycle github-slugger@1.1.2~postinstall: github-slugger@1.1.2
npm sill lifecycle github-slugger@1.1.2~postinstall: no script for postinstall, continuing
npm verb unlock done using /Users/revinguillen/.npm/_locks/staging-525484a240850021.lock for /Users/revinguillen/tmp/tmp/node_modules/.staging
npm sill install saveToDependencies
npm WARN saveError ENOENT: no such file or directory, open '/Users/revinguillen/tmp/tmp/package.json'
npm info lifecycle undefined~preshrinkwrap: undefined
npm sill lifecycle undefined~preshrinkwrap: no script for preshrinkwrap, continuing
npm info lifecycle undefined~shrinkwrap: undefined
npm sill lifecycle undefined~shrinkwrap: no script for shrinkwrap, continuing
npm notice created a lockfile as package-lock.json. You should commit this file.
npm info lifecycle undefined~postshrinkwrap: undefined
npm sill lifecycle undefined~postshrinkwrap: no script for postshrinkwrap, continuing
npm WARN enoent ENOENT: no such file or directory, open '/Users/revinguillen/tmp/tmp/package.json'
npm verb enoent This is related to npm not being able to find a file.
npm verb enoent
npm WARN tmp No description
npm WARN tmp No repository field.
npm WARN tmp No README data
npm WARN tmp No license field.

npm sill install printInstalled
added 1 package in 1.168s
npm verb exit [ 0, true ]
npm info ok
</pre>
</details>

##### Result

```sh
$ npm ls emoji-regex
/Users/revinguillen/tmp/tmp
└─┬ github-slugger@1.1.2
  └── UNMET DEPENDENCY emoji-regex@^>=6.0.0 <=6.1.1

npm ERR! missing: emoji-regex@^>=6.0.0 <=6.1.1, required by github-slugger@1.1.2
```

Having a valid package.json in place (generated with `npm init -y`) doesn't seem to make a difference either.

The real project in which I discovered this is [marky-markdown](https://github.com/npm/marky-markdown); starting from an environment set up with npm 4.6.1, then removing node_modules/ and doing `npm i -g npm@next` to get npm5, then doing `npm i`, I found that `emoji-regex` wasn't installed. `npm i` with 4.6.1 produces a working dependency tree.

### supporting information:

 - `npm -v` prints: **`5.0.0`**
 - `node -v` prints: **`v6.10.0`**
 - `npm config get registry` prints: **`https://registry.npmjs.org/`**
 - Windows, OS X/macOS, or Linux?: **OS X**

### npm 4.6.1 behavior, for reference

<details>
<summary>Click for logging output</summary>
<pre>
npm info it worked if it ends with ok
npm verb cli [ '/Users/revinguillen/.nvm/versions/node/v6.10.0/bin/node',
npm verb cli   '/Users/revinguillen/.nvm/versions/node/v6.10.0/bin/npm',
npm verb cli   'i',
npm verb cli   '-ddd',
npm verb cli   'github-slugger@^1' ]
npm info using npm@4.6.1
npm info using node@v6.10.0
npm sill install loadCurrentTree
npm sill install readLocalPackageData
npm sill fetchPackageMetaData github-slugger@^1
npm sill fetchNamedPackageData github-slugger
npm sill mapToRegistry name github-slugger
npm sill mapToRegistry using default registry
npm sill mapToRegistry registry https://registry.npmjs.org/
npm sill mapToRegistry data Result {
npm sill mapToRegistry   raw: 'github-slugger',
npm sill mapToRegistry   scope: null,
npm sill mapToRegistry   escapedName: 'github-slugger',
npm sill mapToRegistry   name: 'github-slugger',
npm sill mapToRegistry   rawSpec: '',
npm sill mapToRegistry   spec: 'latest',
npm sill mapToRegistry   type: 'tag' }
npm sill mapToRegistry uri https://registry.npmjs.org/github-slugger
npm verb request uri https://registry.npmjs.org/github-slugger
npm verb request no auth needed
npm info attempt registry request try #1 at 12:38:34 PM
npm verb request using bearer token for auth
npm verb request id 03e547da76201495
npm verb etag W/"5928508e-2422"
npm verb lastModified Fri, 26 May 2017 15:58:06 GMT
npm http request GET https://registry.npmjs.org/github-slugger
npm http 304 https://registry.npmjs.org/github-slugger
npm verb headers { server: 'nginx/1.10.1',
npm verb headers   'last-modified': 'Fri, 26 May 2017 15:58:06 GMT',
npm verb headers   etag: '"5928508e-2422"',
npm verb headers   'cache-control': 'max-age=300',
npm verb headers   'accept-ranges': 'bytes',
npm verb headers   date: 'Fri, 26 May 2017 19:38:34 GMT',
npm verb headers   via: '1.1 varnish',
npm verb headers   connection: 'keep-alive',
npm verb headers   'x-served-by': 'cache-sjc3635-SJC',
npm verb headers   'x-cache': 'MISS',
npm verb headers   'x-cache-hits': '0',
npm verb headers   'x-timer': 'S1495827514.250647,VS0,VE116',
npm verb headers   vary: 'Accept-Encoding' }
npm sill get cb [ 304,
npm sill get   { server: 'nginx/1.10.1',
npm sill get     'last-modified': 'Fri, 26 May 2017 15:58:06 GMT',
npm sill get     etag: '"5928508e-2422"',
npm sill get     'cache-control': 'max-age=300',
npm sill get     'accept-ranges': 'bytes',
npm sill get     date: 'Fri, 26 May 2017 19:38:34 GMT',
npm sill get     via: '1.1 varnish',
npm sill get     connection: 'keep-alive',
npm sill get     'x-served-by': 'cache-sjc3635-SJC',
npm sill get     'x-cache': 'MISS',
npm sill get     'x-cache-hits': '0',
npm sill get     'x-timer': 'S1495827514.250647,VS0,VE116',
npm sill get     vary: 'Accept-Encoding' } ]
npm verb etag https://registry.npmjs.org/github-slugger from cache
npm verb get saving github-slugger to /Users/revinguillen/.npm/registry.npmjs.org/github-slugger/.cache.json
npm verb correctMkdir /Users/revinguillen/.npm correctMkdir not in flight; initializing
npm sill pickManifestFromRegistryMetadata spec >=1.0.0 <2.0.0 tag latest versions [ '1.0.0', '1.0.1', '1.1.0', '1.1.1', '1.1.2' ]
npm sill install normalizeTree
npm sill install loadIdealTree
npm sill install cloneCurrentTreeToIdealTree
npm sill install loadShrinkwrap
npm sill install loadAllDepsIntoIdealTree
npm sill resolveWithNewModule github-slugger@1.1.2 checking installable status
npm sill cache add args [ 'github-slugger@^1', null ]
npm verb cache add spec github-slugger@^1
npm sill cache add parsed spec Result {
npm sill cache add   raw: 'github-slugger@^1',
npm sill cache add   scope: null,
npm sill cache add   escapedName: 'github-slugger',
npm sill cache add   name: 'github-slugger',
npm sill cache add   rawSpec: '^1',
npm sill cache add   spec: '>=1.0.0 <2.0.0',
npm sill cache add   type: 'range' }
npm sill addNamed github-slugger@>=1.0.0 <2.0.0
npm verb addNamed ">=1.0.0 <2.0.0" is a valid semver range for github-slugger
npm sill addNameRange { name: 'github-slugger',
npm sill addNameRange   range: '>=1.0.0 <2.0.0',
npm sill addNameRange   hasData: false }
npm sill mapToRegistry name github-slugger
npm sill mapToRegistry using default registry
npm sill mapToRegistry registry https://registry.npmjs.org/
npm sill mapToRegistry data Result {
npm sill mapToRegistry   raw: 'github-slugger',
npm sill mapToRegistry   scope: null,
npm sill mapToRegistry   escapedName: 'github-slugger',
npm sill mapToRegistry   name: 'github-slugger',
npm sill mapToRegistry   rawSpec: '',
npm sill mapToRegistry   spec: 'latest',
npm sill mapToRegistry   type: 'tag' }
npm sill mapToRegistry uri https://registry.npmjs.org/github-slugger
npm verb addNameRange registry:https://registry.npmjs.org/github-slugger not in flight; fetching
npm verb get https://registry.npmjs.org/github-slugger not expired, no request
npm sill addNameRange number 2 { name: 'github-slugger',
npm sill addNameRange   range: '>=1.0.0 <2.0.0',
npm sill addNameRange   hasData: true }
npm sill addNameRange versions [ 'github-slugger',
npm sill addNameRange   [ '1.0.0', '1.0.1', '1.1.0', '1.1.1', '1.1.2' ] ]
npm sill pickManifestFromRegistryMetadata spec >=1.0.0 <2.0.0 tag latest versions [ '1.0.0', '1.0.1', '1.1.0', '1.1.1', '1.1.2' ]
npm sill addNamed github-slugger@latest
npm verb addNamed "latest" is being treated as a dist-tag for github-slugger
npm info addNameTag [ 'github-slugger', 'latest' ]
npm sill mapToRegistry name github-slugger
npm sill mapToRegistry using default registry
npm sill mapToRegistry registry https://registry.npmjs.org/
npm sill mapToRegistry data Result {
npm sill mapToRegistry   raw: 'github-slugger',
npm sill mapToRegistry   scope: null,
npm sill mapToRegistry   escapedName: 'github-slugger',
npm sill mapToRegistry   name: 'github-slugger',
npm sill mapToRegistry   rawSpec: '',
npm sill mapToRegistry   spec: 'latest',
npm sill mapToRegistry   type: 'tag' }
npm sill mapToRegistry uri https://registry.npmjs.org/github-slugger
npm verb addNameTag registry:https://registry.npmjs.org/github-slugger already in flight; waiting
npm sill addNameTag next cb for github-slugger with tag latest
npm sill addNamed github-slugger@1.1.2
npm verb addNamed "1.1.2" is a plain semver version for github-slugger
npm sill cache afterAdd github-slugger@1.1.2
npm verb afterAdd /Users/revinguillen/.npm/github-slugger/1.1.2/package/package.json not in flight; writing
npm verb correctMkdir /Users/revinguillen/.npm correctMkdir not in flight; initializing
npm verb afterAdd /Users/revinguillen/.npm/github-slugger/1.1.2/package/package.json written
npm sill fetchNamedPackageData emoji-regex
npm sill mapToRegistry name emoji-regex
npm sill mapToRegistry using default registry
npm sill mapToRegistry registry https://registry.npmjs.org/
npm sill mapToRegistry data Result {
npm sill mapToRegistry   raw: 'emoji-regex',
npm sill mapToRegistry   scope: null,
npm sill mapToRegistry   escapedName: 'emoji-regex',
npm sill mapToRegistry   name: 'emoji-regex',
npm sill mapToRegistry   rawSpec: '',
npm sill mapToRegistry   spec: 'latest',
npm sill mapToRegistry   type: 'tag' }
npm sill mapToRegistry uri https://registry.npmjs.org/emoji-regex
npm verb request uri https://registry.npmjs.org/emoji-regex
npm verb request no auth needed
npm info attempt registry request try #1 at 12:38:34 PM
npm verb request using bearer token for auth
npm verb etag W/"59093fac-6a90"
npm verb lastModified Wed, 03 May 2017 02:25:48 GMT
npm http request GET https://registry.npmjs.org/emoji-regex
npm http 304 https://registry.npmjs.org/emoji-regex
npm verb headers { server: 'nginx/1.10.3',
npm verb headers   'last-modified': 'Wed, 03 May 2017 02:25:48 GMT',
npm verb headers   etag: '"59093fac-6a90"',
npm verb headers   'cache-control': 'max-age=300',
npm verb headers   'accept-ranges': 'bytes',
npm verb headers   date: 'Fri, 26 May 2017 19:38:34 GMT',
npm verb headers   via: '1.1 varnish',
npm verb headers   connection: 'keep-alive',
npm verb headers   'x-served-by': 'cache-sjc3635-SJC',
npm verb headers   'x-cache': 'MISS',
npm verb headers   'x-cache-hits': '0',
npm verb headers   'x-timer': 'S1495827515.622199,VS0,VE141',
npm verb headers   vary: 'Accept-Encoding' }
npm sill get cb [ 304,
npm sill get   { server: 'nginx/1.10.3',
npm sill get     'last-modified': 'Wed, 03 May 2017 02:25:48 GMT',
npm sill get     etag: '"59093fac-6a90"',
npm sill get     'cache-control': 'max-age=300',
npm sill get     'accept-ranges': 'bytes',
npm sill get     date: 'Fri, 26 May 2017 19:38:34 GMT',
npm sill get     via: '1.1 varnish',
npm sill get     connection: 'keep-alive',
npm sill get     'x-served-by': 'cache-sjc3635-SJC',
npm sill get     'x-cache': 'MISS',
npm sill get     'x-cache-hits': '0',
npm sill get     'x-timer': 'S1495827515.622199,VS0,VE141',
npm sill get     vary: 'Accept-Encoding' } ]
npm verb etag https://registry.npmjs.org/emoji-regex from cache
npm verb get saving emoji-regex to /Users/revinguillen/.npm/registry.npmjs.org/emoji-regex/.cache.json
npm verb correctMkdir /Users/revinguillen/.npm correctMkdir not in flight; initializing
npm sill pickManifestFromRegistryMetadata spec <=6.1.1 tag latest versions [ '1.0.0',
npm sill pickManifestFromRegistryMetadata   '1.0.1',
npm sill pickManifestFromRegistryMetadata   '2.0.0',
npm sill pickManifestFromRegistryMetadata   '3.0.0',
npm sill pickManifestFromRegistryMetadata   '4.0.0',
npm sill pickManifestFromRegistryMetadata   '5.0.0',
npm sill pickManifestFromRegistryMetadata   '6.0.0',
npm sill pickManifestFromRegistryMetadata   '6.1.0',
npm sill pickManifestFromRegistryMetadata   '6.1.1',
npm sill pickManifestFromRegistryMetadata   '6.1.3',
npm sill pickManifestFromRegistryMetadata   '6.2.0',
npm sill pickManifestFromRegistryMetadata   '6.3.0',
npm sill pickManifestFromRegistryMetadata   '6.4.0',
npm sill pickManifestFromRegistryMetadata   '6.4.1',
npm sill pickManifestFromRegistryMetadata   '6.4.2' ]
npm sill resolveWithNewModule emoji-regex@6.1.1 checking installable status
npm sill cache add args [ 'emoji-regex@^>=6.0.0 <=6.1.1', null ]
npm verb cache add spec emoji-regex@^>=6.0.0 <=6.1.1
npm sill cache add parsed spec Result {
npm sill cache add   raw: 'emoji-regex@^>=6.0.0 <=6.1.1',
npm sill cache add   scope: null,
npm sill cache add   escapedName: 'emoji-regex',
npm sill cache add   name: 'emoji-regex',
npm sill cache add   rawSpec: '^>=6.0.0 <=6.1.1',
npm sill cache add   spec: '<=6.1.1',
npm sill cache add   type: 'range' }
npm sill addNamed emoji-regex@<=6.1.1
npm verb addNamed "<=6.1.1" is a valid semver range for emoji-regex
npm sill addNameRange { name: 'emoji-regex', range: '<=6.1.1', hasData: false }
npm sill mapToRegistry name emoji-regex
npm sill mapToRegistry using default registry
npm sill mapToRegistry registry https://registry.npmjs.org/
npm sill mapToRegistry data Result {
npm sill mapToRegistry   raw: 'emoji-regex',
npm sill mapToRegistry   scope: null,
npm sill mapToRegistry   escapedName: 'emoji-regex',
npm sill mapToRegistry   name: 'emoji-regex',
npm sill mapToRegistry   rawSpec: '',
npm sill mapToRegistry   spec: 'latest',
npm sill mapToRegistry   type: 'tag' }
npm sill mapToRegistry uri https://registry.npmjs.org/emoji-regex
npm verb addNameRange registry:https://registry.npmjs.org/emoji-regex not in flight; fetching
npm verb get https://registry.npmjs.org/emoji-regex not expired, no request
npm sill addNameRange number 2 { name: 'emoji-regex', range: '<=6.1.1', hasData: true }
npm sill addNameRange versions [ 'emoji-regex',
npm sill addNameRange   [ '1.0.0',
npm sill addNameRange     '1.0.1',
npm sill addNameRange     '2.0.0',
npm sill addNameRange     '3.0.0',
npm sill addNameRange     '4.0.0',
npm sill addNameRange     '5.0.0',
npm sill addNameRange     '6.0.0',
npm sill addNameRange     '6.1.0',
npm sill addNameRange     '6.1.1',
npm sill addNameRange     '6.1.3',
npm sill addNameRange     '6.2.0',
npm sill addNameRange     '6.3.0',
npm sill addNameRange     '6.4.0',
npm sill addNameRange     '6.4.1',
npm sill addNameRange     '6.4.2' ] ]
npm sill pickManifestFromRegistryMetadata spec <=6.1.1 tag latest versions [ '1.0.0',
npm sill pickManifestFromRegistryMetadata   '1.0.1',
npm sill pickManifestFromRegistryMetadata   '2.0.0',
npm sill pickManifestFromRegistryMetadata   '3.0.0',
npm sill pickManifestFromRegistryMetadata   '4.0.0',
npm sill pickManifestFromRegistryMetadata   '5.0.0',
npm sill pickManifestFromRegistryMetadata   '6.0.0',
npm sill pickManifestFromRegistryMetadata   '6.1.0',
npm sill pickManifestFromRegistryMetadata   '6.1.1',
npm sill pickManifestFromRegistryMetadata   '6.1.3',
npm sill pickManifestFromRegistryMetadata   '6.2.0',
npm sill pickManifestFromRegistryMetadata   '6.3.0',
npm sill pickManifestFromRegistryMetadata   '6.4.0',
npm sill pickManifestFromRegistryMetadata   '6.4.1',
npm sill pickManifestFromRegistryMetadata   '6.4.2' ]
npm sill addNamed emoji-regex@6.1.1
npm verb addNamed "6.1.1" is a plain semver version for emoji-regex
npm sill cache afterAdd emoji-regex@6.1.1
npm verb afterAdd /Users/revinguillen/.npm/emoji-regex/6.1.1/package/package.json not in flight; writing
npm verb correctMkdir /Users/revinguillen/.npm correctMkdir not in flight; initializing
npm verb afterAdd /Users/revinguillen/.npm/emoji-regex/6.1.1/package/package.json written
npm sill currentTree tmp
npm sill idealTree tmp
npm sill idealTree ├── emoji-regex@6.1.1
npm sill idealTree └── github-slugger@1.1.2
npm sill install generateActionsToTake
npm sill diffTrees action count 2
npm sill diffTrees add emoji-regex@6.1.1
npm sill diffTrees add github-slugger@1.1.2
npm sill decomposeActions action count 16
npm sill decomposeActions fetch emoji-regex@6.1.1
npm sill decomposeActions extract emoji-regex@6.1.1
npm sill decomposeActions test emoji-regex@6.1.1
npm sill decomposeActions preinstall emoji-regex@6.1.1
npm sill decomposeActions build emoji-regex@6.1.1
npm sill decomposeActions install emoji-regex@6.1.1
npm sill decomposeActions postinstall emoji-regex@6.1.1
npm sill decomposeActions finalize emoji-regex@6.1.1
npm sill decomposeActions fetch github-slugger@1.1.2
npm sill decomposeActions extract github-slugger@1.1.2
npm sill decomposeActions test github-slugger@1.1.2
npm sill decomposeActions preinstall github-slugger@1.1.2
npm sill decomposeActions build github-slugger@1.1.2
npm sill decomposeActions install github-slugger@1.1.2
npm sill decomposeActions postinstall github-slugger@1.1.2
npm sill decomposeActions finalize github-slugger@1.1.2
npm sill install executeActions
npm sill doSerial global-install 0
npm sill doParallel fetch 2
npm verb correctMkdir /Users/revinguillen/.npm/_locks correctMkdir not in flight; initializing
npm verb lock using /Users/revinguillen/.npm/_locks/staging-525484a240850021.lock for /Users/revinguillen/tmp/tmp/node_modules/.staging
npm sill doParallel extract 2
npm sill extract emoji-regex@6.1.1
npm sill extract github-slugger@1.1.2
npm verb unbuild node_modules/.staging/emoji-regex-64d56544
npm verb unbuild node_modules/.staging/github-slugger-c3f5c079
npm sill gentlyRm /Users/revinguillen/tmp/tmp/node_modules/.staging/emoji-regex-64d56544 is being purged from base /Users/revinguillen/tmp/tmp
npm verb gentlyRm don't care about contents; nuking /Users/revinguillen/tmp/tmp/node_modules/.staging/emoji-regex-64d56544
npm verb tar unpack /Users/revinguillen/.npm/emoji-regex/6.1.1/package.tgz
npm verb tar unpacking to /Users/revinguillen/tmp/tmp/node_modules/.staging/emoji-regex-64d56544
npm sill gentlyRm /Users/revinguillen/tmp/tmp/node_modules/.staging/emoji-regex-64d56544 is being purged
npm verb gentlyRm don't care about contents; nuking /Users/revinguillen/tmp/tmp/node_modules/.staging/emoji-regex-64d56544
npm sill gentlyRm /Users/revinguillen/tmp/tmp/node_modules/.staging/github-slugger-c3f5c079 is being purged from base /Users/revinguillen/tmp/tmp
npm verb gentlyRm don't care about contents; nuking /Users/revinguillen/tmp/tmp/node_modules/.staging/github-slugger-c3f5c079
npm sill gunzTarPerm modes [ '755', '644' ]
npm verb tar unpack /Users/revinguillen/.npm/github-slugger/1.1.2/package.tgz
npm verb tar unpacking to /Users/revinguillen/tmp/tmp/node_modules/.staging/github-slugger-c3f5c079
npm sill gentlyRm /Users/revinguillen/tmp/tmp/node_modules/.staging/github-slugger-c3f5c079 is being purged
npm verb gentlyRm don't care about contents; nuking /Users/revinguillen/tmp/tmp/node_modules/.staging/github-slugger-c3f5c079
npm sill gunzTarPerm modes [ '755', '644' ]
npm sill gunzTarPerm extractEntry package.json
npm sill gunzTarPerm extractEntry package.json
npm sill gunzTarPerm extractEntry README.md
npm sill gunzTarPerm extractEntry index.js
npm sill gunzTarPerm extractEntry .npmignore
npm sill gunzTarPerm extractEntry README.md
npm sill gunzTarPerm extractEntry LICENSE
npm sill gunzTarPerm extractEntry index.js
npm sill gunzTarPerm extractEntry CONTRIBUTING.md
npm sill gunzTarPerm extractEntry .travis.yml
npm sill gunzTarPerm extractEntry CHANGELOG.md
npm sill gunzTarPerm extractEntry test/index.js
npm sill gunzTarPerm extractEntry LICENSE-MIT.txt
npm sill gunzTarPerm extractEntry dist/index.js
npm sill gentlyRm /Users/revinguillen/tmp/tmp/node_modules/.staging/emoji-regex-64d56544/node_modules is being purged
npm verb gentlyRm don't care about contents; nuking /Users/revinguillen/tmp/tmp/node_modules/.staging/emoji-regex-64d56544/node_modules
npm sill gentlyRm /Users/revinguillen/tmp/tmp/node_modules/.staging/github-slugger-c3f5c079/node_modules is being purged
npm verb gentlyRm don't care about contents; nuking /Users/revinguillen/tmp/tmp/node_modules/.staging/github-slugger-c3f5c079/node_modules
npm sill doParallel preinstall 2
npm sill preinstall emoji-regex@6.1.1
npm info lifecycle emoji-regex@6.1.1~preinstall: emoji-regex@6.1.1
npm sill preinstall github-slugger@1.1.2
npm info lifecycle github-slugger@1.1.2~preinstall: github-slugger@1.1.2
npm sill lifecycle emoji-regex@6.1.1~preinstall: no script for preinstall, continuing
npm sill lifecycle github-slugger@1.1.2~preinstall: no script for preinstall, continuing
npm sill doReverseSerial remove 0
npm sill doSerial move 0
npm sill doSerial finalize 2
npm sill finalize /Users/revinguillen/tmp/tmp/node_modules/emoji-regex
npm sill finalize /Users/revinguillen/tmp/tmp/node_modules/github-slugger
npm sill doSerial build 2
npm sill build emoji-regex@6.1.1
npm info linkStuff emoji-regex@6.1.1
npm sill linkStuff emoji-regex@6.1.1 has /Users/revinguillen/tmp/tmp/node_modules as its parent node_modules
npm verb linkBins emoji-regex@6.1.1
npm verb linkMans emoji-regex@6.1.1
npm sill build github-slugger@1.1.2
npm info linkStuff github-slugger@1.1.2
npm sill linkStuff github-slugger@1.1.2 has /Users/revinguillen/tmp/tmp/node_modules as its parent node_modules
npm verb linkBins github-slugger@1.1.2
npm verb linkMans github-slugger@1.1.2
npm sill doSerial global-link 0
npm sill doParallel update-linked 0
npm sill doSerial install 2
npm sill install emoji-regex@6.1.1
npm info lifecycle emoji-regex@6.1.1~install: emoji-regex@6.1.1
npm sill lifecycle emoji-regex@6.1.1~install: no script for install, continuing
npm sill install github-slugger@1.1.2
npm info lifecycle github-slugger@1.1.2~install: github-slugger@1.1.2
npm sill lifecycle github-slugger@1.1.2~install: no script for install, continuing
npm sill doSerial postinstall 2
npm sill postinstall emoji-regex@6.1.1
npm info lifecycle emoji-regex@6.1.1~postinstall: emoji-regex@6.1.1
npm sill lifecycle emoji-regex@6.1.1~postinstall: no script for postinstall, continuing
npm sill postinstall github-slugger@1.1.2
npm info lifecycle github-slugger@1.1.2~postinstall: github-slugger@1.1.2
npm sill lifecycle github-slugger@1.1.2~postinstall: no script for postinstall, continuing
npm verb unlock done using /Users/revinguillen/.npm/_locks/staging-525484a240850021.lock for /Users/revinguillen/tmp/tmp/node_modules/.staging
npm sill install printInstalled
/Users/revinguillen/tmp/tmp
└─┬ github-slugger@1.1.2
  └── emoji-regex@6.1.1

npm WARN enoent ENOENT: no such file or directory, open '/Users/revinguillen/tmp/tmp/package.json'
npm verb enoent This is most likely not a problem with npm itself
npm verb enoent and is related to npm not being able to find a file.
npm verb enoent
npm WARN tmp No description
npm WARN tmp No repository field.
npm WARN tmp No README data
npm WARN tmp No license field.
npm verb exit [ 0, true ]
npm info ok
</pre>
</details>

**Comment by zkat** [comment]
Thanks for the report, Revin! And nice catch. :) It's nice to have people trying out @next and catching these things for us <3 <3 <3 This is pretty 😱

**Comment by lukechilds** [comment]
Getting the same issue with [buttercup-importer](https://github.com/buttercup/buttercup-importer). Normal `npm install` works fine but with `npm@next` or `npm5` the dependency `kdbxweb` is not installed.

Dependencies in package.json:

```js
"dependencies": {
  "buttercup": "~0.38.1",
  "csv-parse": "^1.2.0",
  "is-dir": "~1.0.0",
  "kdbxweb": "~1.0.1",
  "pify": "~2.3.0",
  "xml2js": "~0.4.13"
}
```

**Comment by zkat** [comment]
@revin ...I started looking closer at this, and now I have no idea how this *ever worked*

Look at the semver for it: `emoji-regex@^>=6.0.0 <=6.1.1`. This isn't a valid semver range at all wtf

HOW DID THIS EVER INSTALL??? 😂  We must've had some ridiculous special case somewhere we dropped.

**Comment by zkat** [comment]
@lukechilds yours seems to have to do with git dependencies and shrinkwrap somehow. Thanks for reporting this, too. Hopefully we can figure it out and fix it right quick :)

It's truly amazing all the bizarre little special cases and invalid states that have somehow just worked with npm so far. :s just a matter of slowly making sure all of that is recreated >_<

**Comment by revin** [comment]
@zkat I wondered about that; is it the caret? I'm far from a semver expert so I wasn't quite sure if that was a problem. I can certainly file a bug on that project if that's the real answer here (and you can add to the release notes, "we no longer support broken semver ranges" 😛)

**Comment by zkat** [comment]
@revin the project should fix this but also npm5 needs to have the same behavior if this is something npm4 was able to do. This is technically invalid, yes. Everything should be fine for you once that dep's range is fixed :)

As I still consider this a bug, I'll leave it open until I figure out what dark sorcery npm4 was doing that was spirited away by pacote.

**Comment by iarna** [comment]
```
$ npm -v
5.0.0
$ npm ls emoji-regex
x@1.0.0 x
└─┬ github-slugger@1.1.1
  └── emoji-regex@6.4.2 
$ npm install github-slugger@1.1.2
added 1 package, removed 1 package and updated 1 package in 0.598s
$ npm ls emoji-regex
x@1.0.0 x
└─┬ github-slugger@1.1.2
  └── emoji-regex@6.1.1  invalid
```

**Comment by iarna** [comment]
Ah, so if I start at 1.1.1 then 1.1.2 gets the emoji-regex install. Otherwise I don't. Weird.

**Comment by lukechilds** [comment]
@zkat should I create a separate issue for https://github.com/npm/npm/issues/16744#issuecomment-304423467

**Comment by zkat** [comment]
@lukechilds oh yes! Please :) 👍

**Comment by lukechilds** [comment]
@zkat hmmn, just updated `npm5` and it appears to be working now `¯\_(ツ)_/¯`

**Comment by aqrln** [comment]
@zkat fwiw, in my opinion throwing an error like `ETARGET` on an invalid version range is a perfectly valid solution too, alternatively to mimicking the old behavior :)

What is really weird though, as @iarna has already noted, after installing `emoji-regex@1.1.1` and doing `npm update`, the range is parsed somehow and `emoji-regex` is downgraded to `6.1.1`, but it is moved from the top-level `node_modules` to one inside the `github-slugger` package and shown as invalid in the output of `npm ls`.

**Comment by zkat** [comment]
This seems to be fixed, as I can no longer repro it myself.

@revin Would you mind trying this with the latest canary (`npm i -g npmc@latest && npmc i github-slugger && npmc ls`) and letting me know if it works for you now? If so, let's close this, hm? 👍

**Comment by revin** [comment]
@zkat Yep, looks fixed. github-slugger 1.1.2 was the one with the sketchy dependency spec; it's been fixed in 1.1.3. However:

```shell
$ npmc i github-slugger@1.1.2

+ github-slugger@1.1.2
added 2 packages in 1.142s

$ grep emoji-regex node_modules/github-slugger/package.json
    "emoji-regex": "^>=6.0.0 <=6.1.1"

$ npmc ls
tmp@1.0.0 /Users/revinguillen/tmp/tmp
└─┬ github-slugger@1.1.2
  └── emoji-regex@6.1.1
```

still works even with the weird one. Awesome! Thanks for handling this 🎉 🎉 🎉

**Comment by revin** [comment]
Looks like [29b4deb8](https://github.com/npm/npm/pull/17007/commits/29b4deb8afac69c3388862998090abdb21aa9cb2) is the one that fixed it.

```
> var semver = require('semver')
undefined
> semver.satisfies('6.1.1', '>=^6.0.0 <= 6.1.1')
false
> semver.satisfies('6.1.1', '>=^6.0.0 <= 6.1.1', true)
true
```
